refactor: migrate edit post view to Folio

wescopeland opened the pull request:
Resolves https://github.com/RetroAchievements/RAWeb/issues/2378.

Migrates _editpost.php_ out of _pages-legacy_ and into a Folio view. No functional changes.

Jamiras commented:
I'm going to need some guidance on this. Doing my normal checkout branch, composer install, npm install, and npm run build, I can't even access the forums page:
```
Laravel\Folio\FolioRoutes::get(): Argument #2 ($arguments) must be of type array, int given, called in /var/www/html/vendor/laravel/folio/src/FolioServiceProvider.php on line 75
```
And what's with all the empty `.getkeep` files?

wescopeland commented:
Apologies for that. When I was testing, I was hitting the route directly, but I forgot to update the `route()` call for this page on the topic view. This is now fixed in latest.

I've also removed the .gitkeep files -- these were mostly just placeholders for stuff in _missing-page-redirector.php_.

You should now be able to view topics normally again.

This is the block of code that does things a little differently (and inspired the mess of PRs I opened today):

```php
render(function (View $view, ForumTopicComment $forumTopicComment) {
    if (!Auth::user()->can('update', $forumTopicComment)) {
        return abort('401');
    }
    
    return $view->with([
        'category' => $forumTopicComment->forumTopic->forum->category,
        'forum' => $forumTopicComment->forumTopic->forum,
        'forumTopic' => $forumTopicComment->forumTopic,
    ]);
});
```

This `render()` hook [builds all the data for the page](https://laravel.com/docs/11.x/folio#render-hooks) without needing to connect anything to a `RouteServiceProvider`. It also has access to DI, so we can inject services directly into these (indeed, in some of the PRs I just converted existing controllers to services).

When a `RouteServiceProvider` is used, it actually overrides/disables all of Folio's built-in features, including middleware. My rush of PRs today was to migrate all of our Folio routes to actually use Folio and load data via `render()` hooks.

Jamiras commented on config/missing-page-redirector.php:
```diff
@@ -108,14 +108,14 @@
          * forums
          */
-        '/forum.php' => '/forums/category/{c}',
+        '/forum.php' => '/forums/category/{c}', // TODO what if no category is specified?
```
The way it's implemented, if `c` is not in the query params, it won't get replaced, so it'll try to open `/forums/category/{c}` and fail.

The redirector is not used as long as the legacy blade still exists, so it's not a problem that needs to be addressed now, but the solution is to implement a multi-redirect like I did for ticketmanager.php:
```
        '/forum.php' => [
            'c' => '/forums/category/{c}',
            '' => '/forums',
        ],
```

Jamiras commented on resources/views/pages/forums/post/[ForumTopicComment]/edit.blade.php:
```diff
@@ -45,16 +37,16 @@
                     id="input_compose"
                     label="{{ __res('message', 1)}} "
-                    :value="$foundPost->Payload"
+                    :value="$forumTopicComment->body"
```
How does `forumTopicComment` get passed here? Doesn't it have to be in the `$view->with([` list?

wescopeland replied:
Folio treats all dynamic fields (`[thing]`) as globally-available context values. They can be "peeked" into in `render()`. More docs here: https://laravel.com/docs/11.x/folio#route-model-binding

```php
// [ForumTopicComment] --> $forumTopicComment
render(function (View $view, ForumTopicComment $forumTopicComment) { ... });
```

However, there's no harm in also putting it in `with()`. I also re-added `@props()` for improved documentation.

PopovMark commented:
The issue with the > signs is still happening. Please fix.

wescopeland commented:
@PopovMark You're going to need to be more specific if I'm going to debug the issue.

How exactly is the issue reproduced? When I naively edit posts, I do not see the &lt; or &gt; appearing on either edit or save.

PopovMark commented:
Nvm, the signs don't break anymore when I submit it like that, they still show up properly in the posts.
